### ISSUE TITLE
Buildings with roof:type = gabled  are missing windows, and the roof color randomly changes (premium)

### ISSUE THREAD

**Issue by 3dimka** [vvoovv/blender-geo#525]
Buildings with roof:type = gabled  are missing windows. 
And their roof color randomly changes every time I re-import the map.
If I change the roof:type  to hipped or flat , the issues are no longer reproducing, so it seems that the gabled  roof triggers them.

![image](https://github.com/user-attachments/assets/3b9d7165-b0ff-48fd-bf69-01d324407c65)

![image](https://github.com/user-attachments/assets/af4cafa0-1508-4940-ae30-68c83d606316)

![image](https://github.com/user-attachments/assets/f54a6ee5-e627-4644-91be-c9d2fc8a1a16)

**Comment by vvoovv** [comment]
Hi @3dimka 

The roof color is set randomlly if it is not defined in the OpenStreetMap data.

**Comment by 3dimka** [comment]
> Hi @3dimka
> 
> The roof color is set randomlly if it is not defined in the OpenStreetMap data.

That's true, when the type and color are not set, the attributes are assigned randomly.

However, in my example I set the roof explicitly to gabled (image#1), and then to hipped (image#2)
And when the roof type is gabled there are no premium textures, like they were not implemented for this roof type. And the whole building loses the premium, features when the roof type changed gabled.

**Comment by vvoovv** [comment]
Textures will be applied to the gabled roofs if you uncheck the option _Enable experimental features_ in the addon preferences.

There is a more advanced code in the _dev_ branch that processes the gabled roofs. If you are interested, I can prepare it for the release.

**Comment by hsstud-io** [comment]
Hi, sorry if it`s not related to original thread. Anyone know why the maps that I extracting seems to be cuted on roof? The attached file is from Saint Paul`s Cathedral
![Untitled](https://github.com/user-attachments/assets/6a324895-4005-4e73-838d-94853c663079)

Thanks in advance

**Comment by vvoovv** [comment]
Hi @hsstud-io 

Most like it is not mapped in details in OpenStreetMaps.

**Comment by 3dimka** [comment]
> Textures will be applied to the gabled roofs if you uncheck the option _Enable experimental features_ in the addon preferences.
> 
> There is a more advanced code in the _dev_ branch that processes the gabled roofs. If you are interested, I can prepare it for the release.

I tried importing without the experimental features and realized that I do need the experimental features, for complex hipped roofs and the simplified "import for export" materials. 

What I did as a temporary fix, I replaced the gabled roofs with hipped in the osm xml file. At least i get the textures, even with the incorrect roof types. If the gabled roofs will end up in a premium release or candidate I will gladly try it.

By the way, I found a cute online 3D visualizer which I often use as a reference, in case if you haven't seen it yet:
https://demo.f4map.com/#lat=48.7765719&lon=9.1782849&zoom=19

What it does that Blosm seem to be missing is the fences and beaches. maybe this will give you some ideas and inspiration.
Thanks!
